### PR TITLE
Ts api updates redmine 2367

### PR DESCRIPTION
**Pull request by oschwaldp-oci**
Primarily Updating api for Receive_Message with MessageHeader and providing interactions of how a header can be received and how the information is populated in the message header.  Added test cases to test the header interactions.

### CODE REVIEW

File: FACE/TS_common.hpp [hunk 1]
```diff
@@ -2,12 +2,19 @@
 #define FACE_TS_COMMON_HPP_HEADER_FILE
 #include "common.hpp"
 
+#include <tao/Basic_Types.h>
+#include <tao/VarOut_T.h>
```
Comment (mitza-oci) [inline FACE/TS_common.hpp]: Do we need these?  Would be good to reduce coupling to TAO headers here.

File: dds/FACE/FaceTSS.cpp [hunk 2]
```diff
@@ -369,5 +421,95 @@ FACE::SYSTEM_TIME_TYPE convertDuration(const DDS::Duration_t& duration)
...
 
+FACE::SYSTEM_TIME_TYPE convertTime(const DDS::Time_t& timestamp)
+{
+  return timestamp.nanosec +
+      timestamp.sec * static_cast<FACE::SYSTEM_TIME_TYPE>(NSEC_PER_SEC);
+}
+
+void populate_header_received(FACE::CONNECTION_ID_TYPE connection_id,
+                              DDS::DomainParticipant_var part,
+                              DDS::SampleInfoSeq sinfo,
```
Comment (iamtheschmitzer) [inline dds/FACE/FaceTSS.cpp]: pass by const reference

File: tests/FACE/Header/face_config.ini [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+message_definition_guid=1
+type_name=IDL:Messenger/MessageTypeSupport:1.0
+max_message_size=300
+
+[connection/pub]
+id=1234567890
+domain=3
+direction=source
+topic=Message
+
```
Comment (iamtheschmitzer) [inline tests/FACE/Header/face_config.ini]: add datawriterqos=Message

File: tests/FACE/Header/face_config.ini [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+[connection/sub]
+id=2
+domain=3
+direction=destination
+topic=Message
+
```
Comment (iamtheschmitzer) [inline tests/FACE/Header/face_config.ini]: add datareaderqos=Message

File: dds/FACE/FaceTSS.cpp [hunk 5]
```diff
@@ -210,6 +222,39 @@ void Destroy_Connection(CONNECTION_ID_TYPE connection_id,
...
+                    /*inout*/ FACE::TS::MessageHeader& message_header,
+                    /*in*/    FACE::MESSAGE_SIZE_TYPE /*message_size*/,
+                    /*out*/   FACE::RETURN_CODE_TYPE& return_code)
+{
+  OPENDDS_MAP(FACE::CONNECTION_ID_TYPE, Entities::FaceReceiver)& readers = Entities::instance()->receivers_;
+  if (!readers.count(connection_id)) {
+    return_code = FACE::INVALID_PARAM;
+    return;
+  }
+  if (transaction_id != 0 && transaction_id == readers[connection_id].last_msg_tid) {
```
Comment (iamtheschmitzer) [inline dds/FACE/FaceTSS.cpp]: Comment the significance of transaction_id != 0

**Pull request merged**